### ISSUE TITLE
Navigator lags compared to NavigatorIOS

### ISSUE THREAD

**Issue by ilkarman** [facebook/react-native#12419]
When I try to use _ReactNative.Navigator_ instead of _ReactNative.NavigatorIOS_ I find that there is a lag when the pages update. For example when I render page 3 I still see some remnants of page 2 for a split second. This happens for iOS and Android. I am using react-native 0.41 and have posted both examples below:

**NavigatorIOS :**

Page 1:
```
  render() {
    return (
      <ReactNative.NavigatorIOS
        style={styles.container}
        navigationBarHidden={true}
        initialRoute={{
          title: '',
          component: Page2,
        }}/>
    );
  }
```
Page 2:
```
  handleLanguageResponse(response) {
    this.props.navigator.push({
      title: 'Languages',
      component: Page3,
      passProps: {
        languages: response,
        func: this.onLangChange // Push parent function
      },
    });
  }
```
Page 3:
```
  constructor(props) {
    super(props);
    this.state = {
      dataSource: dataSource.cloneWithRows(this.props.languages)
    };
  }
```

**Navigator:**

Page 1:
```
  renderScene (route, navigator) {
    return <route.component navigator={navigator} data={route.data} />
  }
  render() {
    return (
      <ReactNative.Navigator
        renderScene={this.renderScene.bind(this)}
        initialRoute={{component: Page2}}  
      />
    );
  }
```
Page 2:
```
  handleLanguageResponse(response) {
    this.props.navigator.push({
      component: Page3,
      data: {
        languages: response,
        func: this.onLangChange // Push parent function
      }
    });
  }
```
Page 3:
```
  constructor(props) {
    super(props);
    this.state = {
      dataSource: dataSource.cloneWithRows(this.props.data.languages)
    };
  }
```

**Comment by scarlac** [comment]
It may be easier to understand what kind of lag you're seeing if you do a screen recording. If you're on macOS, you can use QuickTime Player -> File -> New Screen Recroding. Post a link here.

**Comment by ide** [comment]
This lag is understandable because of the threading model (NavIOS uses UI thread + CoreAnimation, Navigator uses just JS thread). This will be addressed with other techniques like native Animated in React Navigation and smoother scheduling with React Fiber.

For now I would recommend using React Navigation and perhaps deferring the rendering of your scenes until the transition is complete.

**Comment by hramos** [comment]
As @ide said this is understandable. While Navigator is not yet officially deprecated, we're encouraging people to start using React Navigation instead. I'm going to close this, but feel free to open a similar issue against React Navigation if needed.

**Comment by keyurparalkar** [comment]
Is this issue resolved in react native : 42.0
